landa/organization_management/doctype/member_function/member_function.py

```diff
@@ -102,6 +103,32 @@ def is_inactive(self):
 		return self.end_date and date_diff(today(), self.end_date) > 0
 
 
+	def validate_unique_roles(self):
+		unique_roles = frappe.get_value(
+			"Member Function Category", self.member_function_category, "only_one_per_organization"
+		)
+
+		member_function_category_name = frappe.get_value(
+			"Member Function Category", self.member_function_category, "name"
+		)
+
+		if unique_roles == 1:
+			existing_member_functions = frappe.get_all(
+				"Member Function",
+				filters={
+					"organization": self.organization,
+					"member": self.member,
+					"member_function_category": self.member_function_category,
+					"name": ["!=", self.name],  # Exclude this document from the search
+				},
+			)
```

Comment:
1. It's best to avoid nesting as much as we can. So we can `return`/exit the function quickly if the role does not have uniqueness.
2. It's enough that we find just 1 other member function and `frappe.db.exists` gives us True when the query returns 1 such value instead of getting all valid values which we do not use.

Pardon the formatting, it often gets ruined in github reviews😅
```suggestion
		if not unique_role:
			return

		existing_member_functions = frappe.db.exists(
			"Member Function",
			{
				"organization": self.organization,
				"member": self.member,
				"member_function_category": self.member_function_category,
				"name": ["!=", self.name],  # Exclude this document from the search
			},
		)
```